### DISCUSSION TITLE
ESLint Rule to Disallow Object Rest Destructuring in `tanstack-query`

### DISCUSSION THREAD

**Discussion by alan910127** [TanStack/query#6265] [Ideas]
## Description

I'd like to propose an enhancement for `@tanstack/eslint-plugin-query` with an ESLint rule, possibly named `no-rest-destructuring`, which discourages or provides an auto-fix for object rest destructuring on the return value of the `useQuery`, `useQueries` and `useInfiniteQuery` hooks. 

## Motivation

As highlighted in [TkDodo's Blog](https://tkdodo.eu/blog/react-query-render-optimizations#tracked-queries), using object rest destructuring tracks all the fields of the query, potentially leading to excessive re-renders. This behavior could impact users of `tanstack-query`.

## Example

```typescript
// the proposed rule would generate an error for this line
const { data, ...rest } = useQuery(...);
```

**Comment by TkDodo** [comment]
Good idea. I would put the level to warning though. @Newbie012 what do you think?

**Reply by Newbie012** [reply]
Sounds like a great idea. If it's okay with you, I'll be happy to implement this rule.
